src/libraries/Common/src/System/Data/Common/DbConnectionOptions.Common.cs

```diff
@@ -114,9 +115,9 @@ private string UsersConnectionString(bool hidePassword, bool forceHidePassword)
 
         public bool ConvertValueToBoolean(string keyName, bool defaultValue)
         {
-            string value;
+            string? value;
             return _parsetable.TryGetValue(keyName, out value) ?
-                ConvertValueToBooleanInternal(keyName, value) :
+                ConvertValueToBooleanInternal(keyName, value!) :
```

Comment:
Is this a bug? i.e. could _parsetable actually contain a null value and then ConvertValueToBooleanInternal will null ref?  If yes, might be worth a comment (I understand your goal is not to fix such bugs now, which is fine)

Reply:
I'm not quite sure anymore, but I think _parseTable might not actually contain nullable values in this codepath, so this could be fine - will add a TODO to investigate though.